helpers/compile/build.ts

```diff
@@ -66,9 +67,16 @@ const applyCjsDefaults = (options: BuildOptions): BuildOptions => ({
   // outfile has precedence over outdir, hence these ternaries
   outdir: options.outfile ? undefined : getOutDir(options),
   // we only produce typescript types on the second run (cjs)
-  plugins: [...(options.plugins ?? []), tscPlugin, onErrorPlugin],
+  plugins: [...(options.plugins ?? []), tscPlugin, replacePlugin, onErrorPlugin],
 })
 
+// because we compile tree-shaken esm to cjs, we need to replace __require
+const replacePlugin = replaceWithPlugin([
+  [/__require\(/g, 'require('], // replace the util calls
+  [/typeof __require/g, 'typeof require'], // replace the util calls
+  [/var __require =.*?(?=var)/gs, ''], // remove the utility
```

Comment:
I wonder if it might be a tiny bit safer to replace the helper with `var __require = require` without replacing `__require` at call sites to minimize the risk of false positives. No strong opinion either way, and the current approach has the advantage of shaving off more extra bytes from the bundle.

Reply:
Yeah, my fear was that some bundlers won't be capable of collapsing the variable and thus skip static analysis. But I don't have enough knowledge with other bundlers other than esbuild or webpack. WDYT?

Reply:
Oh, I see. I think it's a valid concern, yes. I don't know for sure if it will actually cause issues with any bundlers, but it sounds like it might.

Reply:
A good thing is that `esbuild` would replace any other `__require` (eg. in another bundled lib/dep) by `__require2`. So we don't have a high risk of accidentally breaking something.

Reply:
That being said, our current e2e does not test for the linked issue. Basically, we have bundling tests but we never bundle the prisma lib (it's marked as external :trollface:)